src/react/components/VectorStyle/VectorStyleModal.jsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { VectorNameInput } from 'oskari-ui/components/VectorStyle';
+import { LocaleConsumer, Controller } from 'oskari-ui/util';
+import { Message, Modal } from 'oskari-ui';
+import { StyleEditor } from 'oskari-ui/components/StyleEditor';
+
+export const VectorStyleModal = LocaleConsumer(({editorState, onCancel, onModalOk, setEditorState, nameValidation, setName}) => {
+    return (
+        <Modal
+            visible={ editorState.modalVisibility }
+            okButtonPros={ 'disabled' }
+            onOk={ () => onModalOk() }
+            onCancel={ onCancel }
+            cancelText={ <Message messageKey="cancel" /> }
+            okText={ <Message messageKey="save" /> }
+        >
+            <VectorNameInput
+                styleName={ editorState.styleName }
+                isValid={ nameValidation(editorState.styleName) }
```

Comment:
I think name validation (non-empty) can be baked in to this component and setName + nameValidation props removed. The onChange for name could be the setEditorState() with the updated name. Do you see some problem with this kind of simplification?